tests/system/action/projector/test_previous.py

```diff
@@ -0,0 +1,88 @@
+from tests.system.action.base import BaseActionTestCase
+
+
+class ProjectorPrevious(BaseActionTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.set_models(
+            {
+                "meeting/1": {},
+                "projector/2": {"meeting_id": 1},
+                "projector/3": {
+                    "current_projection_ids": [1, 2],
+                    "preview_projection_ids": [3, 4],
+                    "history_projection_ids": [6, 7],
+                    "meeting_id": 1,
+                },
+                "projector/4": {
+                    "current_projection_ids": [],
+                    "preview_projection_ids": [],
+                    "history_projection_ids": [5],
+                    "meeting_id": 1,
+                },
+                "projection/1": {
+                    "current_projector_id": 3,
+                    "meeting_id": 1,
+                    "weight": 100,
+                    "stable": True,
+                },
+                "projection/2": {
+                    "current_projector_id": 3,
+                    "meeting_id": 1,
+                    "weight": 98,
+                },
+                "projection/3": {
+                    "preview_projector_id": 3,
+                    "meeting_id": 1,
+                    "weight": 99,
+                },
+                "projection/4": {
+                    "preview_projector_id": 3,
+                    "meeting_id": 1,
+                    "weight": 100,
+                },
+                "projection/5": {
+                    "history_projector_id": 4,
+                    "meeting_id": 1,
+                    "weight": 100,
+                },
+                "projection/6": {
+                    "history_projector_id": 3,
+                    "meeting_id": 1,
+                    "weight": 60,
+                },
+                "projection/7": {
+                    "history_projector_id": 3,
+                    "meeting_id": 1,
+                    "weight": 55,
+                },
+            }
+        )
+
+    def test_next_nothing(self) -> None:
+        response = self.request("projector.previous", {"id": 2})
+        self.assert_status_code(response, 200)
+        projector = self.get_model("projector/2")
+        assert projector.get("current_projection_ids") is None
+        assert projector.get("preview_projection_ids") is None
+        assert projector.get("history_projection_ids") is None
+
+    def test_next_complex(self) -> None:
+        response = self.request("projector.previous", {"id": 3})
+        self.assert_status_code(response, 200)
+        projector = self.get_model("projector/3")
+        assert projector.get("current_projection_ids") == [6, 1]
+        assert projector.get("preview_projection_ids") == [2, 3, 4]
+        assert projector.get("history_projection_ids") == [7]
+        projection_1 = self.get_model("projection/1")
```

Comment:
this check is not relevant I think - the weight for current projections doesn't affect anything.

Reply:
So removed.